EngineLayer/ProteinScoringAndFdr/ProteinScoringAndFdrEngine.cs

```diff
@@ -48,10 +48,14 @@ private void ScoreProteinGroups(List<ProteinGroup> proteinGroups, IEnumerable<Pe
                     {
                         foreach (var pepWithSetMods in psm.CompactPeptides.SelectMany(b => b.Value.Item2))
                         {
-                            if (!peptideToPsmMatching.TryGetValue(pepWithSetMods, out HashSet<PeptideSpectralMatch> psmsForThisPeptide))
-                                peptideToPsmMatching.Add(pepWithSetMods, new HashSet<PeptideSpectralMatch> { psm });
-                            else
-                                psmsForThisPeptide.Add(psm);
+                            if (pepWithSetMods.DigestionParams.Protease == psm.DigestionParams.Protease)
```

Comment:
compact peptide does not have any digestion parameters associated with it. So I cant check before paring. How it is set up right now, all PWSM that have the same compact peptide are linked regardless of digestion parameters, and then this has to be handled elsewhere. I know this will all change once compact peptide is phased out. PSM only has digestion parameters because I wanted to be able to associate PSM with protease a long time ago, which is coming in handy now.